sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/internal/CheckpointMarkImpl.java

```diff
@@ -62,7 +66,11 @@ public CheckpointMarkImpl decode(InputStream inStream) throws IOException {
 
   @Override
   public void finalizeCheckpoint() {
-    checkState(committer.isPresent());
-    committer.get().commitOffset(offset);
+    try {
+      checkState(committer.isPresent());
```

Comment:
finalizeCheckpoint call is best effort but does the current implementation rely on this being invoked exactly ?
For example, will the reader read duplicate messages if a finalizeCheckpoint fails ?